### PR TITLE
DietPi-Software | Move Stretch instances from MySQL to MariaDB

### PR DESCRIPTION
**Pull request by MichaIng**
https://github.com/Fourdee/DietPi/pull/1276

### CODE REVIEW

File: dietpi/patch_file [hunk 1]
```diff
@@ -2941,6 +2941,34 @@ _EOF_
 
 		fi
 		#-------------------------------------------------------------------------------
+		if [ grep 'aSOFTWARE_INSTALL_STATE[86]=2' /Dietpi/dietpi/.installed ] && (( $DISTRO > 3 )); then
```
Comment (Fourdee) [inline dietpi/patch_file]: You'll need to escape the ```[]```, and correct ```/DietPi/dietpi/.installed```
```
root@DietPi:~# grep 'aSOFTWARE_INSTALL_STATE\[74\]=0' /DietPi/dietpi/.installed
aSOFTWARE_INSTALL_STATE[74]=0

```

Comment (MichaIng) [reply dietpi/patch_file]: @Fourdee 
Fixed and tested. Now the syntax works.

How to set if grep into brackets? [ ] and (( )) are not allowed here. Left it without now, which works, but especially with more than one condition, brackets would enhance the overview.

**Comment by Fourdee** [comment]
@MichaIng 

Many thanks 👍

**Comment by MichaIng** [comment]
@Fourdee 
I just faced the following:
- After applying the patchfile, thus moving from mysql to mariadb, apt package switch works fine, mariadb directly does its job and dietpi-software correctly shows mariadb + webserver stacks installed instead of mysql.
- But if I uninstall mariadb via dietpi-software ui, it does NOT uninstall it. Seems somehow the script goes around it. Do we need to adjust the installed flags somewhere else for this to work?
- Running `dietpi-software uninstall 88` instead works fine, so seems to bypass the issue.

**Comment by MichaIng** [comment]
@Fourdee 
About the dependency list:
- I just wrote a little script that shows installed dependants of all installed packages. With this we can further reduce our dependency list to the final packages that are really directly used by dietpi / usual system operations. Here your dependency list where I added `🈯️ > dependant`, if one within our anyway installed packages exist:
		'acl' 🈺 > systemd on Jessie, but not on Stretch+Buster, so we really need this, `getfacl` or `setfacl`
		'adduser' 🈯️ > systemd+apt+cron+udev+several other packages, depending on distro
		'apt' 🈯️ > apt-utils+apt-transport-https
		'apt-transport-https'
		'apt-utils'
		'base-files' 🈯️ > bash
		'base-passwd' 🈯️ essential package will be never autoremoved
		'bash' 🈯️ > bash-completion + essential
		'bash-completion'
		'bc'
		'bsdmainutils' 🈺 Do we need this? Was not in the list before!
		'bsdutils' 🈯️ essential
		'bzip2'
		'ca-certificates'
		'console-common' 🈺 `console-setup` supersedes or depends on other console related packages
		'console-data' 🈺 `console-setup` supersedes or depends on other console related packages
		'console-setup'
		'console-setup-linux' 🈺 `console-setup` supersedes or depends on other console related packages
		'coreutils' 🈯️ essential
		'cpio' 🈯️ (> initramfs-tools-core) > initramfs-tools
		'crda'
		'cron'
		'curl'
		'dash' 🈯️ > bash > bash-completion
		'dbus' 
		'debconf' 🈯️ > debconf-utils
		'debian-archive-keyring' 🈯️ > apt and wrong on Raspbian, where raspbian-archive-keyring takes place!
		'debianutils' 🈯️ > cron+bash+dash+some others, depending on distro
		'diffutils' 🈯️ essential
		'dmidecode' 🈺 Do we need this? Was not in the list before!
		'dmsetup' 🈯️ > > > parted
		'dosfstools' 🈺 Could be left out on VMs, as we do not have fat boot partition there.
		'dphys-swapfile'
		'dpkg' 🈯️ > cron+bash-completion+others+anyway essential
		'e2fslibs' 🈯️ > e2fsprogs
		'e2fsprogs' 🈯️ essential
		'ethtool'
		'fake-hwclock'
		'fbset'
		'findutils' 🈯️ essential
		'firmware-atheros' 🈺 was x86 only before?
		'firmware-brcm80211' 🈺 was x86 only before?
		'firmware-ralink' 🈺 was x86 only before?
		'firmware-realtek' 🈺 was x86 only before?
		'firmware-misc-nonfree' 🈺 was x86 only before?
		'fuse' 🈺 Didn't see this before, do we need it? https://packages.debian.org/de/stretch/fuse
		'gnupg' 🈺 We used this before, but `gpgv` seems to be a lightweight version of it.
		'gpgv' 🈺 Superseded by `gnupg`, but more lightweight, maybe we need to switch?
		'grep' 🈯️ essential
		'groff-base' 🈺 Do we need this? Was not in the list before!
		'gzip' 🈯️ essential
		'hdparm'
		'hfsplus'
		'hostname' 🈯️ essential
		'htop'
		'ifupdown' 🈯️ > resolvconf
		'init' 🈯️ essential
		'init-system-helpers' 🈯️ > init and essential
		'initramfs-tools'
		'initscripts' 🈺 Was dependency of systemd on Jessie, but since Stretch not anymore. I guess systemd now supersedes that package.
		'insserv' 🈺 same than initscripts, don't have any of them on my systems (besides Jessie VM) and absolutely no issues.
		'iproute2' 🈯️ > ifupdown > resolvconf
		'iputils-ping' 🈺 Helpful indeed sometimes, but not necessary? Also superseded by curl, wget etc?
		'isc-dhcp-client'
		'isc-dhcp-common' 🈯️ This is just man pages!
		'iw'
		'kbd' 🈺 `console-setup` supersedes or depends on other console related packages
		'keyboard-configuration' 🈺 `console-setup` supersedes or depends on other console related packages
		'klibc-utils' 🈯️ initramfs-tools-core > initramfs-tools
		'kmod' 🈯️ (> initramfs-tools-core) > initramfs-tools
		'less' 🈺 Helpful indeed sometimes, but not necessary?
		'locales'
		'login' 🈯️ essential
		'lsb-base' 🈯️ > cron+udev+swapfile+resolvconf+others
		'mawk' 🈯️ > dash > bash
		'mount' 🈯️ > systemd
		'multiarch-support' 🈯️ > hundrets of libs, also e2fslibs > e2fsprogs, which is essential
		'nano' not necessary, but yeah I use this several times a day, everybody needs a simple editor!
		'ncurses-base' 🈯️ essential
		'ncurses-bin' 🈯️ essential
		'net-tools'
		'ntfs-3g'
		'ntp'
		'p7zip-full'
		'parted'
		'passwd' 🈯️ > adduser
		'perl-base' 🈯️ > init-system-helpers+debconf > debconf-utils
		'procps' 🈯️ > udev+systemd
		'psmisc'
		'readline-common' 🈯️ > parted+gnu(pg)
		'resolvconf'
		'rfkill' 			#Used by some onboard WiFi chipsets
		'rsync' 			🈯️ Not necessary here, as it will be installed on demand by dietpi-backup/sync
		'sed' 🈯️ essential
		'sensible-utils' 🈺 Dependency of debianutils until buster. Just wondered that it is not autoremoved on my buster VM (marked as auto), but no dependency anymore and not essential. Just purged it on my Buster VM without any warnings or additional removed packages. Let's see what happens 😄.
		'startpar' 🈺 Was dependency of sysv-rc > systemd on Jessie, but since Stretch, systemd does not depend on sysv-rc anymore. Seems to solve the tasks differently now (same as insserv)?
		'sudo'
		'systemd' 🈯️ > systemd-sysv > init
		'systemd-sysv' 🈯️ > init
		'sysvinit-utils' 🈯️ essential
		'tar' 🈯️ essential
		'tzdata'
		'udev' 🈯️ > initramfs-tools-core > initramfs-tools
		'unzip'
		'usbutils'
		'util-linux' 🈯️ > udev+systemd+others+essential
		'wget'
		'whiptail'
		'wireless-regdb' 🈯️ > crda
		'wireless-tools'
		'wpasupplicant'
		'wput'
		'zip'

By removing the green ones from the list, there should be no additional packages removed. By removing also some of the orange ones, you get the list I used in my PR: https://github.com/Fourdee/DietPi/pull/1266/files#diff-e9e0c6c64b4937739e13ffac58f4a888R74

**Comment by Fourdee** [comment]
@MichaIng 

> But if I uninstall mariadb via dietpi-software ui, it does NOT uninstall it. Seems somehow the script goes around it. Do we need to adjust the installed flags somewhere else for this to work?

I'll check

I'll move your post regarding prep into the other ticket: https://github.com/Fourdee/DietPi/issues/1285#issuecomment-351888843

**Comment by Fourdee** [comment]
@MichaIng 

Fixed uninstall issue: https://github.com/Fourdee/DietPi/commit/1611ee6b787f07ad475bfa3375573db3ae6f6e8a. Smallest bug fix ever (1 char)? :rofl:

**Comment by MichaIng** [comment]
@Fourdee 
Nice! Okay, so this was not related to the PR.

And thx for moving the comment. Seems I switch in between while checking automated script and found issue on VM cleaning 😆.

**Pull request merged**